WalletConnect/SubInteractor/WCBinanceInteractor.swift

```diff
@@ -0,0 +1,36 @@
+// Copyright © 2017-2019 Trust Wallet.
+//
+// This file is part of Trust. The full Trust copyright notice, including
+// terms governing use, modification, and redistribution, is contained in the
+// file LICENSE at the root of the source code distribution tree.
+
+import Foundation
+import PromiseKit
+
+public typealias BnbSignClosure = (_ id: Int64, _ order: WCBinanceOrder) -> Void
+
+public struct WCBinanceInteractor {
+    public var onSign: BnbSignClosure?
+
+    var confirmResolvers: [Int64: Resolver<WCBinanceTxConfirmParam>] = [:]
+
+    mutating func handleEvent(_ event: WCEvent, topic: String, decrypted: Data) throws {
+        switch event {
+        case .bnbSign:
+            if let request: JSONRPCRequest<[WCBinanceTradeOrder]> = try? event.decode(decrypted) {
+                onSign?(request.id, request.params[0])
```

Comment:
did you consider switch case?

Reply:
since we don't know the order type yet, so we have to try decoding to each model, how to apply `switch` in this case?

Reply:
Maybe: https://medium.com/makingtuenti/indeterminate-types-with-codable-in-swift-5a1af0aa9f3d this will work? we currently use a similar approach in the app for `TransactionMetadata`, can you check if that could be applicable?

Reply:
it seems not easy, because the difference is deeply nested in `params`, no explicit order type to map into a enum